app/code/core/Mage/Adminhtml/Block/Widget/Grid.php

```diff
@@ -458,16 +483,20 @@ protected function _setFilterValues($data)
         return $this;
     }
 
+    /**
+     * @param Mage_Adminhtml_Block_Widget_Grid_Column $column
+     * @return $this
+     */
     protected function _addColumnFilterToCollection($column)
     {
         if ($this->getCollection()) {
-            $field = ( $column->getFilterIndex() ) ? $column->getFilterIndex() : $column->getIndex();
+            $field = ($column->getFilterIndex()) ?: $column->getIndex();
             if ($column->getFilterConditionCallback() && $column->getFilterConditionCallback()[0] instanceof self) {
                 call_user_func($column->getFilterConditionCallback(), $this->getCollection(), $column);
             } else {
                 $cond = $column->getFilter()->getCondition();
-                if ($field && isset($cond)) {
-                    $this->getCollection()->addFieldToFilter($field , $cond);
+                if ($field) {
```

Comment:
Is this the same as the original `if ($field && isset($cond)) {`?

Reply:
`$cond` is set one line before as an array ... so `isset($cond)` is useless.

Reply:
But its value can be `null`. No?

Reply:
From the [manual](https://www.php.net/manual/en/function.isset):

> isset() will return false when checking a variable that has been assigned to null

So, `isset($cond)` is not useless.